packages/js/src/ui/helpers/utils.ts

```diff
@@ -25,6 +25,14 @@ function generateUniqueRandomString(set: Set<string>, length: number): string {
   return randomString;
 }
 
+export function getNestedProperty<T>(obj: T, path: string): any {
```

Comment:
💡 We might flatten the localization and appearance JSON for faster runtime access so that the frequent access during rendering is always O(1). 

This can be done as an optimization in a separate PR.